test: add tests to cover new Exceptions in arguments.py

qlrd opened the pull request:
Add tests for arguments.py

qlrd committed:
refactor(arguments): specify ArgumentError exceptions for invalid fingerprints

qlrd committed:
test(arguments): add tests to cover new Exceptions in `arguments.py`

These tests cover, indirectly, the new exceptions in `test_descriptor`
(covering wrong fingerprints lenghts) as well in `test_taproot`
(covering a try to tweak taproot in a non taproot generated key from a
wpkh descriptor).

odudex commented on src/embit/descriptor/arguments.py:
```diff
@@ -14,9 +14,11 @@ def __init__(self, fingerprint: bytes, derivation: list):
     @classmethod
     def from_string(cls, s: str):
         arr = s.split("/")
+        if len(arr[0]) % 2 != 0:
```
I believe this check aims to avoid unhandled exceptions with `unhexlify`, right? Maybe `unhexlify` could be inside a `try/except` to also catch other invalid strings and non hex characters.
Also, remembering this lib is made to run on limited devices, I would reduce verbosity to a minimum:
"Invalid fingerprint format"
"Invalid fingerprint length"

qlrd replied:
Fair, this will allow another tests

odudex commented on src/embit/descriptor/arguments.py:
```diff
@@ -14,7 +14,10 @@ def __init__(self, fingerprint: bytes, derivation: list):
+        try:
+            mfp = unhexlify(arr[0])
+        except Exception as err:
+            raise ArgumentError(err) from err
```
Just to be sure, did you test in device if `raise from` is functional in micropython?
If it was not used so for by Krux or Embit, it would be good to test.

qlrd replied:
Mmm, good point. Didn't tested. My bad for this lack of attention. I believe `micropython` do not supoort it right? Maybe something like the below can be better?

```suggestion
            raise ArgumentError(str(err))
```

odudex replied:
Maybe it works the way it is, but it needs to be tested

qlrd replied:
now that you said, maybe these are theoretical cases, but I believe that in real devices, they may not happen. Maybe it's a dead-code?

qlrd replied:
tested in a `wondermv`, but did not get any of these theoretical errors (IMO because the devices, in a real situations, will never load incorrect fingerprints).

qlrd committed:
refactor(descriptor): add try/except check for key, miniscript and taptree

The current `Descriptor` class do not verify if the instances are
comming from `Key`, `Miniscript` or `TapTree`. This commit add these
checks and raise exceptions when needed.

qlrd committed:
test(descriptor): add descriptor tests for failure during initialization
